interface/ceed-operator.c

```diff
@@ -1425,6 +1425,30 @@ int CeedOperatorGetFlopsEstimate(CeedOperator op, CeedSize *flops) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get CeedQFunction global context for a CeedOperator.
+           Note: The caller is responsible for destroying `ctx` via `CeedQFunctionContextDestroy()`.
+           Note: If `*ctx` is non-NULL, then it is assumed that `*ctx` is a pointer to a CeedQFunctionContext.
+             This CeedQFunctionContext will be destroyed if `*ctx` is the only reference to this CeedQFunctionContext.
```

Comment:
This seems redundant and a little confusing. I see that the second note is posted on other functions, but usually by itself. Having both kinda confused me here that there might be some special rules with the function, but they're both talking about the same idea more-or-less. I think the first one is more direct (the second basically explains the concept of the `*Destroy` functions).

Reply:
I think there are three ideas here:

1) If the `ctx` passed in is non-null, then its assumed to be actually be a `ctx` and it will be destroyed as part of this function call (so `ctx` passed in should hold an actual QFunctionContext or should be initialized to NULL)

2) If that `ctx` is the last reference to a CeedQFunctionContext, then that memory will be freed

3) The `ctx` that is passed back out does need to be destroyed

Reply:
Oooh, gotcha. I thought `*ctx` was referring to the state after the function execution, not the state being passed into the function.

Reply:
Tweaked the wording a bit to hopefully make this clearer

Reply:
Yep, reads clear to me now!

Reply:
Might be worth changing the other uses of that phrase too, assuming it also applies. Grepping `non-NULL` did the trick for me bringing them all up.